web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts

```diff
@@ -129,8 +139,24 @@ export function useCreateDatabase() {
       .fetchDatabases(clusterId, request, signal)
       .then(res => {
         if (res.agents.length) {
-          return res.agents[0];
+          const dbServer = res.agents[0];
+          if (
+            !isAws ||
+            !dbServer.aws?.iamPolicyStatus ||
+            dbServer.aws?.iamPolicyStatus === IamPolicyStatus.Unspecified
+          ) {
+            return dbServer;
+          }
+
+          if (dbServer.aws.iamPolicyStatus !== IamPolicyStatus.Pending) {
+            return dbServer;
+          }
```

Comment:
I believe these are equivalent:
```suggestion
          const dbServer = res.agents[0];
          if (
            !isAws || // If not AWS, then we return the first thing we get back.
            dbServer.aws?.iamPolicyStatus !== IamPolicyStatus.Pending // If AWS and aws.iamPolicyStatus is undefined or non-pending, we return the dbServer.
          ) {
            return dbServer;
          }
          // Otherwise continue polling
```

Reply:
done